feature_engine/transformation/log.py

```diff
@@ -164,3 +164,206 @@ def _more_tags(self):
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of numerical variables to transform. If None, the transformer
+        will find and select all numerical variables.
+
+    base: string, default='e'
+        Indicates if the natural or base 10 logarithm should be applied. Can take
+        values 'e' or '10'.
+
+    C: "auto" or int, default="auto"
+        The constant C to apply the transfomation log(x + C).
+
+        - If int, then log(x + C)
+        - If "auto", then C = abs(min(x)) + 1
+
+    Attributes
+    ----------
+    variables_:
+        The group of variables that will be transformed.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    C_:
+        The constant C to transform data.
+
+    Methods
+    -------
+    fit:
+        This transformer learns the constant C to add to log transformation.
+    transform:
+        Transforms the variables using log transformation.
+    fit_transform:
+        Fit to data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        base: str = "e",
+        C: Union[int, str] = "auto",
+    ) -> None:
+
+        if base not in ["e", "10"]:
+            raise ValueError("base can take only '10' or 'e' as values")
+
+        if not isinstance(C, int) and not C == "auto":
+            raise ValueError("C can take only 'auto' or int")
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.base = base
+        self.C = C
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer learns the constant C to add to log transformation.
+
+        Select the numerical variables and determines whether the logarithm
+        can be applied on the selected variables (it checks if the variables
+        are all positive after the addition of C).
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            - If the input is not a Pandas DataFrame
+            - If any of the user provided variables are not numerical
+        ValueError
+            - If there are no numerical variables in the df or the df is empty
+            - If the variable(s) contain null values
+            - If some variables contain zero or negative values after suming constant C
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        # calculate C to add to each variable
+        if self.C == "auto":
+            self.C_ = X[self.variables_].min(axis=0).abs() + 1
```

Comment:
This is perhaps a painful request because it might slow things down, but I'll make it anyhow and we can discuss about it.

All Feature-engine transformers store learn parameters when they differ per variable as a dictionary, so that the user can easily see which value corresponds to each variable. And this is a key difference with sklearn arrays, where you can't really understand without some data manipulation what value corresponds to what.

I find this very useful. So I would like C_ to be a dictionary of variable: minimum +1 value pairs. Could we do that?

The display in rst would also need some changing after this modification

Reply:
will add this also as input option for `C` param 

```python
C: "auto", int or dict, default="auto"
    The constant C to apply the transfomation log(x + C).

      - If int, then log(x + C)
      - If "auto", then C = abs(min(x)) + 1
      - If dict, dictionary mapping the constant C to apply
        to each variable.
```

Reply:
makes total sense, great idea